Node/Managing Node-NPM versions.md

```diff
@@ -0,0 +1,126 @@
+
+[Node version manager](https://github.com/nvm-sh/nvm) is a version manager similar to `n` and a good alternative. You can create a `.nvmrc` file containing a Node version number. Running `use`, `install`, `which` etc. commands will use the version specified in the file if none was provided in the command line. When using `nvm` and installing global packages, those packages will only be global for that specific version. eg. You currently use Node 16 and install `polyglot-cli`, if you switch to v18, the package won't be available there until you install it.
+
+```bash
+#Install NVM by running script
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash
+
+#Install latest Node version
+nvm install node
+
+#Install specific version of Node by specifying version or use an alias
+nvm install <version>
+nvm install <alias>
+
+#Check Node version currently in use and available versions
+nvm ls
+
+#Set default version
+nvm alias default <version>
+
+#Switching between Node versions
+nvm use <version>
+nvm use default
+
+#Uninstall Node version
+nvm uninstall <version>
+```
+
+#### Using homebrew
+
+For this method you need to have [homebrew](https://brew.sh/) installed on your Mac(which you probably do). There might be additional steps required when switching versions using `brew` as switching is not supported, so you might need to uninstall and unlink the old version, and install and link new one. Using this method for managing Node versions is discouraged as it is likely that issues will occur at some point if multiple versions are needed.
+
+```bash
+#Install Node
+brew install node
+
+#Install specific version of node
+brew install node@<version>
+
+#Update Node
+brew upgrade node
+
+#Uninstall Node
+brew uninstall node
+```
+
+#### Using direct binary install
+
+Considering previously mentioned options, there is no particular advantage why you would use binary installation. You can still opt in to do so if you wish.
+
+#### avn - Automatic Version Switching
+
+[avn](https://github.com/wbyoung/avn) is a package for automatic version switching of Node using one of your Node version manager package. Currently supported version managing packages are `n`, `nvm` and `nodebrew`. This package is not mandatory and will only work in conjunction with one of the version managing tools. When you `cd` into a directory containing `.node-version` file, `avn` will automatically detect the change and use version manager of your choice to switch to the specified version. Given the `.node-version` and `avn` support, it would be recommended to use `n` as version manager.
+
+```bash
+#Install the packages
+npm install -g avn avn-nvm avn-n
+avn setup
+```
+
+#### Specifying version in package.json
+
+You can specify the version of node that your stuff works on by providing Node versions in package.json. If you don't specify this, it means any version will do.
+
+```json
+{
+  "engines": {
+    "node": ">= 18.12"
```

Comment:
We should also write down npm version here, as minor NPM versions can also be breaking changes (*cough* [`npm v8.4.1`](https://infinum.slack.com/archives/C01MJ2HDA05/p1655806084190259) *cough*)